pkg/agent/openflow/pipeline_windows.go

```diff
@@ -190,11 +200,78 @@ func (c *client) snatImplementationFlows(nodeIP net.IP, category cookie.Category
 	return flows
 }
 
+// serviceSNATFlows installs flows to allow a Windows host to access a ClusterIP Service.
+func (c *client) serviceSNATFlows(nodeIP net.IP, localGatewayMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+	if c.enableProxy {
+		flows = append(flows, []binding.Flow{
+			// If the host uses one IP other than Node management IP to access the Service, the Service reply
+			// packet needs unSNAT with snatCTMark mark.
+			c.pipeline[uplinkTable].BuildFlow(priorityNormal).
+				MatchProtocol(binding.ProtocolIP).
+				MatchRegRange(int(marksReg), markTrafficFromUplink, trafficFromRegRange).
+				Action().CT(false, unSNATTable, ctZoneSNAT).NAT().CTDone().
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			c.pipeline[unSNATTable].BuildFlow(priorityNormal).
+				MatchProtocol(binding.ProtocolIP).
+				MatchCTMark(snatCTMark, nil).
+				MatchCTStateRpl(true).
+				MatchCTStateTrk(true).
+				Action().SetDstMAC(localGatewayMAC).
+				Action().GotoTable(conntrackTable).
+				Done(),
+			c.pipeline[unSNATTable].BuildFlow(priorityLow).
+				MatchRegRange(int(marksReg), markTrafficFromUplink, trafficFromRegRange).
+				Action().Output(config.BridgeOFPort).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			// Flows below are for hostnetwork ClusterIP Service which needs SNAT and the SNAT mark is set.
```

Comment:
Nit: hostnetwork -> hostNetwork

Reply:
And it can also be Services with external endpoints?

Reply:
Updated `hostNetwork` here and other places in logs and comments.